### ISSUE TITLE
How to create custom

### ISSUE THREAD

**Issue by Prasannajnaeyulu** [CANopenNode/CANopenLinux#1]

**Comment by CANopenNode** [comment]
Yes, you are right. You can also check https://github.com/CANopenNode/CANopenDemo/tree/master/demo

**Comment by Prasannajnaeyulu** [comment]
Ok. But the Object Dictionary editor from this library is not working with our can device eds file it is throwing errors. Whereas other editors able to load the file properly ( say, vector caneds is able to load successfully but it doesn't has capability to export OD.c and OD.h files). I have to make this library Object Dictionary works with our eds files so that I am able to generate required OD.c and OD.h files. Any Idea what's wrong with this library Object Dictionary?
![image](https://user-images.githubusercontent.com/2508680/123869591-18545a80-d929-11eb-8f55-1f14a97d4144.png)


But I couldn't see any problem with the entry in .eds file
```
[2003sub1]
ParameterName=Valve1
ObjectType=0x8
DataType=0x0001
AccessType=rwr
DefaultValue=0
ObjFlags=1
Description="Control of the valve"```

**Comment by CANopenNode** [comment]
I'm not the author of the eds parser, I'm not sure about the problem.

Please check the parser here:
https://github.com/CANopenNode/CANopenEditor/blob/2e8a14a17a7ee183225e8df7958908ce1f2b0b5f/libEDSsharp/eds.cs#L2333

**Comment by Prasannajnaeyulu** [comment]
Thanks for the reference. I will look into that.

**Comment by CarmenMFM** [comment]
Hello!
I have created  a CANOpen node using our own Object Dictionary using EDSEditor.
However, when i put OD.h and OD.c in CANOpennode/example and sudo make in linux terminal i have next error.

![ERROR_OD](https://user-images.githubusercontent.com/92932039/153203381-73f3f09d-d2c8-4aed-b85b-df6788366e57.jpeg)

Can someone please help me? Thanks!

**Comment by AndreaRoberti** [comment]
Hi all, 
We are facing the same error described by @CarmenMFM . Any update? 
We notice that importing our EDS file with the editor, the generated files do not contain the define macro (OD_CNT_HB_PROD or _EM_PROD)

Thanks in advance

**Comment by CANopenNode** [comment]
There is no automation, when importing own EDS file. CANopenNode requires some custom properties to be set, especially in communication section. It is best to use original CANopenNode project file and add own objects. See https://github.com/CANopenNode/CANopenNode/tree/master#object-dictionary-editor

**Comment by OmarAtomica** [comment]
Hello i have created my own Od.c & Od.h using the object dictionary editor from my .eds file however i still don't understand how can i use the canopen library with other bits of my code for example i want to build an application layer that utilizes the features of the canOpen node to operate a motor controller , so how can i integrate these codes together ?

**Comment by CANopenNode** [comment]
Please see CANopenDemo.
